rpc/types/src/json.rs

```diff
@@ -86,6 +95,558 @@ define_request_and_response! {
     }
 }
 
+define_request_and_response! {
+    get_block_count,
+    cc73fe71162d564ffda8e549b79a350bca53c454 =>
+    core_rpc_server_commands_defs.h => 919..=933,
+    GetBlockCount,
+
+    // There are no request fields specified,
+    // this will cause the macro to generate a
+    // type alias to `()` instead of a `struct`.
+    Request {},
+
+    #[derive(Copy)]
+    ResponseBase {
+        count: u64,
+    }
+}
+
+define_request_and_response! {
+    on_get_block_hash,
+    cc73fe71162d564ffda8e549b79a350bca53c454 =>
+    core_rpc_server_commands_defs.h => 935..=939,
+    OnGetBlockHash,
+    /// ```rust
+    /// use serde_json::*;
+    /// use cuprate_rpc_types::json::*;
+    ///
+    /// let x = OnGetBlockHashRequest { block_height: [3] };
+    /// let x = to_string(&x).unwrap();
+    /// assert_eq!(x, "[3]");
+    /// ```
+    #[cfg_attr(feature = "serde", serde(transparent))]
+    #[repr(transparent)]
+    #[derive(Copy)]
+    Request {
+        // This is `std::vector<u64>` in `monerod` but
+        // it must be a 1 length array or else it will error.
+        block_height: [u64; 1],
+    },
+    /// ```rust
+    /// use serde_json::*;
+    /// use cuprate_rpc_types::json::*;
+    ///
+    /// let x = OnGetBlockHashResponse { block_hash: String::from("asdf") };
+    /// let x = to_string(&x).unwrap();
+    /// assert_eq!(x, "\"asdf\"");
+    /// ```
+    #[cfg_attr(feature = "serde", serde(transparent))]
+    #[repr(transparent)]
+    Response {
+        block_hash: String,
+    }
+}
+
+define_request_and_response! {
+    submit_block,
+    cc73fe71162d564ffda8e549b79a350bca53c454 =>
+    core_rpc_server_commands_defs.h => 1114..=1128,
+    SubmitBlock,
+    /// ```rust
+    /// use serde_json::*;
+    /// use cuprate_rpc_types::json::*;
+    ///
+    /// let x = SubmitBlockRequest { block_ids: vec!["a".into(),"b".into()] };
+    /// let x = to_string(&x).unwrap();
+    /// assert_eq!(x, r#"["a","b"]"#);
+    /// ```
+    #[cfg_attr(feature = "serde", serde(transparent))]
+    #[repr(transparent)]
+    Request {
+        block_ids: Vec<String>,
```

Comment:
This is a block blob:

```suggestion
        // This is `std::vector<std::string>` in `monerod` but
        // it must be a 1 length array or else it will error.
        block_blob: [String; 1],
```

Reply:
Yup: https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server.cpp#L2222-L2234, the output only returns 1 block too so yeah.

Documentation unfortunately states:
> array of strings; list of block **blobs** which have been mined

FWIW I expect many optimizations like this to made to the types when the handlers are created and we see how the types are used.